shark/examples/shark_inference/stable_diffusion/utils.py

```diff
@@ -86,3 +89,117 @@ def set_iree_runtime_flags():
+                key_combination (int, optional): choice for mapping value for device name.
+                1 : path
+                2 : name
+                3 : (name, path)
+                Defaults to 3.
+            Returns:
+                dict: map to possible device names user can input mapped to desired combination of name/path.
+            """
+            from shark.iree_utils._common import iree_device_map
+
+            driver = iree_device_map(driver)
+            device_list = get_all_devices(driver)
+            device_map = dict()
+
+            def get_output_value(dev_dict):
+                if key_combination == 1:
+                    return f"{driver}://{dev_dict['path']}"
+                if key_combination == 2:
+                    return dev_dict["name"]
+                if key_combination == 3:
+                    return (dev_dict["name"], f"{driver}://{dev_dict['path']}")
+
+            # mapping driver name to default device (driver://0)
+            device_map[f"{driver}"] = get_output_value(device_list[0])
+            for i, device in enumerate(device_list):
+                # mapping with index
+                device_map[f"{driver}://{i}"] = get_output_value(device)
+                # mapping with full path
+                device_map[f"{driver}://{device['path']}"] = get_output_value(
+                    device
+                )
+            return device_map
+
+        def map_device_to_name_path(device, key_combination=3):
+            """Gives the appropriate device data (supported name/path) for user selected execution device
+            Args:
+                device (str): user
+                key_combination (int, optional): choice for mapping value for device name.
+                1 : path
+                2 : name
+                3 : (name, path)
+                Defaults to 3.
+            Raises:
+                ValueError:
+            Returns:
+                str / tuple: returns the mapping str or tuple of mapping str for the device depending on key_combination value
+            """
+            driver = device.split("://")[0]
+            device_map = get_device_mapping(driver, key_combination)
+            try:
+                device_mapping = device_map[device]
+            except KeyError:
+                raise ValueError(f"Device '{device}' is not a valid device.")
+            return device_mapping
+
+        name, args.device = map_device_to_name_path(args.device)
+        triple = get_vulkan_target_triple(name)
+        print(f"Found device {name}. Using target triple {triple}")
+        # set triple flag to avoid multiple calls to get_vulkan_triple_flag
+        if args.iree_vulkan_target_triple == "" and triple is not None:
+            args.iree_vulkan_target_triple = triple
+
+        # use tuned models only in the case of rdna3 cards.
+        if not args.iree_vulkan_target_triple:
+            vulkan_triple_flags = get_vulkan_target_triple(name)
```

Comment:
Why are we doing this twice? We can just look at the previous result from get_vulkan_target_triple.

Reply:
Done